src/System.Private.CoreLib/shared/System/Threading/EventWaitHandle.Windows.cs

```diff
@@ -0,0 +1,95 @@
+        {
+#if !PLATFORM_WINDOWS
+            if (name != null)
+                throw new PlatformNotSupportedException(SR.PlatformNotSupported_NamedSynchronizationPrimitives);
+#endif
+            uint eventFlags = initialState ? Interop.Kernel32.CREATE_EVENT_INITIAL_SET : 0;
+            if (mode == EventResetMode.ManualReset)
+                eventFlags |= (uint)Interop.Kernel32.CREATE_EVENT_MANUAL_RESET;
+
+            SafeWaitHandle handle = Interop.Kernel32.CreateEventEx(IntPtr.Zero, name, eventFlags, AccessRights);
+
+            int errorCode = Marshal.GetLastWin32Error();
+            if (handle.IsInvalid)
+            {
+                handle.SetHandleAsInvalid();
+                if (name != null && name.Length != 0 && errorCode == Interop.Errors.ERROR_INVALID_HANDLE)
+                    throw new WaitHandleCannotBeOpenedException(SR.Format(SR.Threading_WaitHandleCannotBeOpenedException_InvalidHandle, name));
+
+                throw Win32Marshal.GetExceptionForWin32Error(errorCode, name);
+            }
+            createdNew = errorCode != Interop.Errors.ERROR_ALREADY_EXISTS;
+            SafeWaitHandle = handle;
+        }
+
+        private static OpenExistingResult OpenExistingWorker(string name, out EventWaitHandle result)
+        {
+#if PLATFORM_WINDOWS
+            if (name == null)
+                throw new ArgumentNullException(nameof(name));
+            if (name.Length == 0)
+                throw new ArgumentException(SR.Argument_EmptyName, nameof(name));
+
+            result = null;
+            SafeWaitHandle myHandle = Interop.Kernel32.OpenEvent(AccessRights, false, name);
+
+            if (myHandle.IsInvalid)
+            {
+                int errorCode = Marshal.GetLastWin32Error();
+
+                if (errorCode == Interop.Errors.ERROR_FILE_NOT_FOUND || errorCode == Interop.Errors.ERROR_INVALID_NAME)
+                    return OpenExistingResult.NameNotFound;
+                if (errorCode == Interop.Errors.ERROR_PATH_NOT_FOUND)
+                    return OpenExistingResult.PathNotFound;
+                if (name != null && name.Length != 0 && errorCode == Interop.Errors.ERROR_INVALID_HANDLE)
+                    return OpenExistingResult.NameInvalid;
+
+                throw Win32Marshal.GetExceptionForWin32Error(errorCode, name);
+            }
+            result = new EventWaitHandle(myHandle);
+            return OpenExistingResult.Success;
+#else
+            throw new PlatformNotSupportedException(SR.PlatformNotSupported_NamedSynchronizationPrimitives);
+#endif
+        }
+
+        public bool Reset()
+        {
+            bool res = Interop.Kernel32.ResetEvent(_waitHandle);
+            if (!res)
+                throw Win32Marshal.GetExceptionForLastWin32Error();
+            return res;
+        }
+        
+        public bool Set() => Set(_waitHandle);
+
+        internal static bool Set(SafeWaitHandle handle)
```

Comment:
It looks like the static method is only used from the Unix threadpool. Another option would be to not have it at all, and just call the low level method (`WaitSubsystem.SetEvent`) from the threadpool, without going through the EventWaitHandle. It would make the shared partition code cleaner.

Reply:
Right, it is the supported build configs today.